<div style="background-color:#3c3d41;padding:30px;border-radius:5px">

<h1 style="font-size:22px;color:white;margin-top:0px">
How to download NASA Earth Science Data from LAADS DAAC using python ?
</h1>

<p style="font-size:14px;color:white">
Author: Benjamin H.G Marchant<br>
License: CC BY 4.0
</p>
  
</div>

The NASA LAADS DAAC (Level-1 and Atmosphere Archive & Distribution System Distributed Active Archive Center) provides satellite data products to users worldwide. These data products support various scientific and educational research projects. In this section, we will explore the methods for locating and downloading data from NASA LAADS DAAC using Python.

Table of Content:

* [Creating an account on NASA LAADS DAAC ](#create_token)
* [Classification of NASA Earth Science Data Products](#classification)
* [Downloading Aqua/MODIS L1 Data](#download_modis_l1)
    * [What is Aqua ?](#aqua_def)
    * [ What is MODIS instrument ?](#modis_def)
    * [Import Necessary Libraries](#necessary_libraries)
    * [Define Inputs Parameters](#inputs_parameters)
    * [Calculate Day of the Year](#DOY)
    * [Set Up Authorization](#Authorization)
    * [Get all file names for a specific day](#request_file_names)
    * [Get file name for a specific time](#find_name)
    * [Download a file from NASA LAADS DAAC](#download)
    * [Create a python script to download a file from NASA](#python_script)
* [Another example: download all files for a specific product and day](#download_files_script)
* [Links](#references)

## Creating an account on NASA LAADS DAAC <a class="anchor" id="create_token"></a>

To download files from NASA LAADS DAAC, you need to create an access token. Follow these steps to generate your token:

Visit the [NASA LAADS DAAC website](https://urs.earthdata.nasa.gov/oauth/authorize?response_type=code&client_id=A6th7HB-3EBoO7iOCiCLlA&redirect_uri=https%3A%2F%2Fladsweb.modaps.eosdis.nasa.gov%2Fcallback&state=U8PNHrqwoeL033HkWCn5DiHPtzeYAc).

- Create a free account by signing up.

- Once logged in, click on your profile icon in the top right corner and select "Generate Token" from the drop-down menu.

- Copy the generated token and keep it secure.

Replace the placeholder token with your own token in your python code. For example:

In [1]:
token = 'YourGeneratedTokenHere'

Here is a sample token for illustration purposes:

In [2]:
token = 'eyJ0eXAiOiJKV1QiLCJvcmlnaW4iOiJFYXJ0aGRhdGEgTG9naW4iLCJzaWciOiJlZGxqd3RwdWJrZXlfb3BzIiwiYWxnIjoiUlMyNTYifQ.eyJ0eXBlIjoiVXNlciIsInVpZCI6ImJtYXJjaGFudCIsImV4cCI6MTcyMjAzOTU1MiwiaWF0IjoxNzE2ODU1NTUyLCJpc3MiOiJFYXJ0aGRhdGEgTG9naW4ifQ.22EbP-kQdHunpQvnSB9eENEeV06J2lhCvliBIPo560VsUm4dBZeQEz-sE8n681zLKUERxOeXqyTWWYPGEtyJeMNuMfYkdAfFZN18VoU2SRG4W3YNanUXTjq15EONDAqyKr-UO-PcV5QKC_Xtkdj8nHlCwIkbzlZIBCBl1jZob0xmNM9qvCeqk0Rk8SXK80nvEz1Ji0fRwbuS_Cl4lgkNKZwMBNdziFP3rVClPews8waKoZmeu0zk20uylYazZY73QN8ZL6-vDYnSZL4Wo7RPsxornLg8wCD_xV4vv5PXzUe82VsthpyiYD03iu93MDOI9qX5lATChY8fo_4kjmkdNw'

Remember to replace 'YourGeneratedTokenHere' with the token you obtained from the NASA LAADS DAAC website.

NASA Earth science data products are categorized into different levels based on their processing and complexity. Here's a general overview:

## Classification of NASA Earth Science Data Products <a class="anchor" id="classification"></a>

**Level 1 (L1)**:

- Level 1 data are usually raw or minimally processed satellite observations.These data often undergo initial processing to correct for sensor-specific effects like geometric distortions, radiometric calibration, and sensor artifacts.
- Level 1 data typically contain basic parameters such as radiance or reflectance values.

**Level 2 (L2)**:

- Level 2 data are derived geophysical variables or intermediate products generated from Level 1 data.
These data often involve more advanced processing algorithms to convert raw measurements into meaningful geophysical parameters.
- Level 2 data may include products such as surface temperature, vegetation indices, atmospheric profiles, or ocean chlorophyll concentration. They provide information that is closer to what scientists need for research and applications.

**Level 3 (L3)**:

- Level 3 data are typically gridded or spatially aggregated products derived from Level 2 data. These data are often averaged or aggregated over specific time intervals (e.g., daily, monthly, yearly) and/or spatial resolutions (e.g., global grids, regional tiles).
- Level 3 data facilitate easier analysis and visualization of long-term trends, spatial patterns, and regional variability. They are commonly used for large-scale studies, climate monitoring, and model validation.

## Downloading Aqua/MODIS L1 Data  <a class="anchor" id="download_modis_l1"></a>

NASA LAADS DAAC is a treasure trove of Earth science data, housing products from various missions and instruments. In this tutorial, we'll take a hands-on approach by downloading a specific type of data: an Aqua/MODIS Level 1 (L1) granule (The MODIS Level 1B data set contains calibrated and geolocated radiances at-aperture for all 36 MODIS spectral bands at 1km resolution. These data are generated from the MODIS Level 1A scans of raw radiance and in the process converted to geophysical units of W / (m^2 um sr)) [ref](https://ladsweb.modaps.eosdis.nasa.gov/missions-and-measurements/products/MOD021KM).

This example will serve as a blueprint for you to explore and download other datasets available on LAADS DAAC.

### What is Aqua ?  <a class="anchor" id="aqua_def"></a>

Aqua is a major international Earth Science satellite mission developed by NASA, launched in 2002. It circles the Earth in a sun-synchronous orbit, meaning it passes over the equator at roughly the same time each day. Aqua carries six instruments that continuously monitor Earth's water cycle, atmosphere, and land masses.

Here's a breakdown of Aqua's key aspects:

- Focus: Earth's water cycle, atmosphere, and land
- Launched by: NASA (in collaboration with international partners)
- Orbit: Sun-synchronous
- Instruments: Six instruments collecting data on various aspects of Earth's systems

### What is MODIS instrument ? <a class="anchor" id="modis_def"></a>

The Moderate Resolution Imaging Spectroradiometer (MODIS) is a key instrument onboard two NASA Earth science satellites:

- Terra: Launched in 1999, Terra circles the Earth in a morning orbit, collecting data around 10:30 AM local time.
- Aqua: Launched in 2002, Aqua follows a sun-synchronous afternoon orbit, acquiring data at approximately 1:30 PM local time.

MODIS: A Powerful Eye on Earth

MODIS acts like a giant camera, but instead of capturing visible light like a regular camera, it detects electromagnetic radiation across a wide range of wavelengths. This allows scientists to observe and measure various aspects of Earth's system, including:

- Land: Vegetation cover, land surface temperature, and changes in deforestation.
- Oceans: Ocean color, which can reveal phytoplankton abundance and ocean health.
- Atmosphere: Cloud cover, aerosols, and atmospheric water vapor.

The following image depicts a full-day daytime observation captured by MODIS:

In [3]:
from IPython.display import Image
from IPython.core.display import HTML 

Image(url= "https://en.moonbooks.org/media/images/webp/modis-daytime-obs.webp?lastmod=1716574684.297826")


The image here has been created using all the observations from one day. However, Level 1 and Level 2 data are not stored in a single file, as it would be too large. Instead, the data are split into smaller chunks known as granules or tiles. Each granule represents 5 minutes of observation for MODIS. Here is an example of a MODIS granule.

In [4]:
Image(url= "https://en.moonbooks.org/media/images/webp/download.webp?lastmod=1716909495.605649")

Here's a detailed explanation of how to download a single granule from NASA DAAC data using Python:

### Import Necessary Libraries <a class="anchor" id="necessary_libraries"></a>

In [5]:
from datetime import date
from os import path

import urllib.request
import urllib.request, json 
import pprint

- datetime: Used to handle date calculations.
- os: Used to handle file path operations.
- urllib.request: Used for making HTTP requests.
- json: Used to parse JSON responses.
- pprint: Used to pretty-print JSON data for readability.

### Define Inputs Parameters <a class="anchor" id="inputs_parameters"></a>

In [6]:
year = 2015
month = 7
day = 18

product_name = 'MYD021KM'
collection = 61

- year, month, day: Define the date for the data request.
- product_name: Specifies the satellite product name (e.g., 'MYD021KM').
- collection: Indicates the collection number for the dataset.

### Calculate Day of the Year <a class="anchor" id="DOY"></a>

While traditional calendar days are sometimes used for data retrieval, Earth science data often employs a more efficient system – the day of year (DOY). This value represents the sequential number of a day within a specific year. For instance, January 1st is DOY 1, and December 31st is DOY 365 (or 366 in leap years).

Benefits of DOY:

- Standardization: DOY eliminates confusion caused by varying month lengths. It ensures consistency across datasets, regardless of the month.
- Efficiency: Calculations involving time become simpler when using DOY.

Example of how to calculate the Day of the Year

In [7]:
d0 = date(year, 1, 1)
d1 = date(year, month, day)
delta = d1 - d0

count_of_day = delta.days + 1

print(count_of_day)

199


- d0: Represents January 1st of the specified year.
- d1: Represents the target date for the data request.
- delta: Calculates the difference between the target date and January 1st.
- count_of_day: Computes the day of the year (DOY) for the target date.

### Set Up Authorization <a class="anchor" id="Authorization"></a>

In [8]:
opener = urllib.request.build_opener()
opener.addheaders = [('Authorization', 'Bearer {}'.format(token))]
urllib.request.install_opener(opener)

This code configures a custom URL opener to automatically include an Authorization header with a Bearer token for all HTTP requests. This setup is useful when accessing APIs or web services that require authentication.

- build_opener(): Creates an opener object to handle HTTP requests.
- addheaders: Adds the authorization header using the token for authenticated access to NASA LAADS DAAC.
- install_opener(opener): Installs the opener as the default opener for urllib.

### Get all file names for a specific day <a class="anchor" id="request_file_names"></a>

Now that you have authorization to send requests to NASA using your token, you can query NASA to retrieve the filenames for the MYD021KM product available on a specific day.

In [9]:
%%time

ladsweb_url = 'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/{}/{}/{:04d}/{:03d}.json'.format(collection, product_name,year,count_of_day)

with urllib.request.urlopen(ladsweb_url) as url:
    data = json.loads(url.read().decode())    

CPU times: user 17.8 ms, sys: 7.37 ms, total: 25.2 ms
Wall time: 1.74 s


The code will return a dictionary called data, which contains all the filenames for the specified day, along with additional information. Let's print the first two entries as an example:

In [10]:
data['content'][:2]

[{'archiveSets': 61,
  'cksum': '2967257851',
  'dataDay': '2015-199 = 2015-07-18',
  'downloadsLink': 'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD021KM/2015/199/MYD021KM.A2015199.0000.061.2018051065758.hdf',
  'fileId': 3066991188,
  'md5sum': '0d22a3542c93d20cb9e4d330ef2eaef5',
  'mtime': 1528501338,
  'name': 'MYD021KM.A2015199.0000.061.2018051065758.hdf',
  'products': 'MYD021KM',
  'resourceType': 'File',
  'self': '/archive/allData/61/MYD021KM/2015/199/MYD021KM.A2015199.0000.061.2018051065758.hdf',
  'size': 164712505},
 {'archiveSets': 61,
  'cksum': '771041275',
  'dataDay': '2015-199 = 2015-07-18',
  'downloadsLink': 'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/61/MYD021KM/2015/199/MYD021KM.A2015199.0005.061.2018051065801.hdf',
  'fileId': 3066991189,
  'md5sum': '2edb5bafadd1412897498ed12b0fbec1',
  'mtime': 1528501338,
  'name': 'MYD021KM.A2015199.0005.061.2018051065801.hdf',
  'products': 'MYD021KM',
  'resourceType': 'File',
  'self': '/archive

Here is an example of how to extract the filename for the first entry:

In [11]:
data['content'][0]['name']

'MYD021KM.A2015199.0000.061.2018051065758.hdf'

The MODIS NASA file naming convention encodes essential metadata about the file within its name. Here is a breakdown of the example filename MYD021KM.A2015199.0000.061.2018051065758.hdf:

- MYD021KM:

- MYD: Indicates the MODIS instrument on the Aqua satellite (as opposed to MOD for Terra).
021KM: Refers to the specific product type, which in this case is the Level-1B calibrated radiances at 1 km resolution.
A2015199:

- A: Stands for Aqua (similarly, T would stand for Terra).
- 2015: The year the data was collected.
- 199: The Julian day of the year (day 199 of 2015, which corresponds to July 18, 2015).

- 0000: Represents the hour and minute of the start of the data collection in Coordinated Universal Time (UTC), formatted as HHMM. In this case, 0000 means 00:00 UTC.

- 061: Indicates the collection version number of the data product. 061 is the collection number, signifying a specific reprocessing campaign.

- 2018051065758: This timestamp (YYYYDDDHHMMSS) indicates when the file was produced. 2018 is the year, 051 is the Julian day (February 20, 2018), 065758 represents the time in HHMMSS (06:57:58 UTC).

- hdf: The file extension indicating the format of the file. HDF stands for Hierarchical Data Format, which is commonly used for storing and distributing scientific data.

In [12]:
type(data)

dict

To check the number of available files, we can use the Python function len(). Here's how we do it:

In [13]:
len(data['content'])

288

This makes sense because, as previously mentioned, a MODIS granule corresponds to 5 minutes. Therefore:

24 hours × 60 minutes ÷ 5 minutes per granule = 288 files

So, if the number of files is lower than 288, it indicates that some MODIS files are missing!

In [14]:
#pp = pprint.PrettyPrinter(indent=4)
#pp.pprint(data)

### Get file name for a specific time <a class="anchor" id="find_name"></a>

Now, let's say we want to find the granule corresponding to a specific date and time.

In [15]:
hour = 19
minute = 35

Note that the minutes must be a multiple of 5 since MODIS granules are generated every 5 minutes. To find the filenames, we need to iterate over all the files because it's not possible to know in advance when each file has been created.

In [16]:
file_name = ''
for file in data['content']:
    #print(file['name'])
    file_name_prefix = '{}.A{:04d}{:03d}.{:02d}{:02d}'.format(product_name,year,count_of_day,hour,minute)
    if file_name_prefix in file['name']:
        file_name = file['name']      
        
print(file_name)

MYD021KM.A2015199.1935.061.2018051071000.hdf


### Download a file from NASA LAADS DAAC <a class="anchor" id="download"></a>

Now that we have the filename, we can use Python to download it:

In [17]:
if not path.exists(file_name):
    ladsweb_url = 'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/{}/{}/{:04d}/{:03d}/{}'.format(collection,product_name,year,count_of_day,file_name)
    urllib.request.urlretrieve(ladsweb_url, file_name) 

### Create a python script to download a file from NASA <a class="anchor" id="python_script"></a>

Let's develop Python functions that encapsulate all the steps outlined above:

In [18]:
from datetime import date
from os import path

import urllib.request
import urllib.request 
import json
import pprint


def get_count_of_day(year,month,day):
    
    """
    Function that returns the count of day (since the beginning of the year) 
    for a given year, month and day
    
    Positional argument:
        year -- type = integer
        month -- type = integer 
        day -- type = integer
    
    Example:
        get_count_of_day(2015,7,18)
        returns: 199
    """
    
    d0 = date(year, 1, 1)
    d1 = date(year, month, day)
    delta = d1 - d0

    count_of_day = delta.days + 1

    return count_of_day
    
def nasa_laads_daac_order_file(product,collection,year,month,day,hour,minute):
    """
    Download a single file from NASA LAADS DAAC 
    (Level-1 and Atmosphere Archive & Distribution System Distributed Active Archive Center)
    (https://ladsweb.modaps.eosdis.nasa.gov/)
    
    Positional argument:
        product -- name of the product (eg. MYD06_L2); type = string
        collection -- collection (eg. 61); type = integer 
        year -- product year (eg. 61); type = integer 
        month -- product month (eg. 61); type = integer 
        day -- product day (eg. 61); type = integer 
        hour -- product hour (eg. 61); type = integer 
        minute -- product minute (eg. 61); type = integer 
    
    Example:
        nasa_laads_daac_order_file("MYD021KM",61,2015,7,18,19,35)
        
    Requirements:
        This function depends on the get_count_of_day function. 
    """
    
    # Step1: Get NASA LAADS DAAC token"
    
    #with open('config.json') as json_data:
    #    config_data = json.load(json_data)
    #    print(config_data)
    
    token = 'eyJ0eXAiOiJKV1QiLCJvcmlnaW4iOiJFYXJ0aGRhdGEgTG9naW4iLCJzaWciOiJlZGxqd3RwdWJrZXlfb3BzIiwiYWxnIjoiUlMyNTYifQ.eyJ0eXBlIjoiVXNlciIsInVpZCI6ImJtYXJjaGFudCIsImV4cCI6MTcyMjAzOTU1MiwiaWF0IjoxNzE2ODU1NTUyLCJpc3MiOiJFYXJ0aGRhdGEgTG9naW4ifQ.22EbP-kQdHunpQvnSB9eENEeV06J2lhCvliBIPo560VsUm4dBZeQEz-sE8n681zLKUERxOeXqyTWWYPGEtyJeMNuMfYkdAfFZN18VoU2SRG4W3YNanUXTjq15EONDAqyKr-UO-PcV5QKC_Xtkdj8nHlCwIkbzlZIBCBl1jZob0xmNM9qvCeqk0Rk8SXK80nvEz1Ji0fRwbuS_Cl4lgkNKZwMBNdziFP3rVClPews8waKoZmeu0zk20uylYazZY73QN8ZL6-vDYnSZL4Wo7RPsxornLg8wCD_xV4vv5PXzUe82VsthpyiYD03iu93MDOI9qX5lATChY8fo_4kjmkdNw'
    
    #print(token)
    
    # Step2: Convert (month,day) to count of day"
    
    count_of_day = get_count_of_day(year,month,day)
    
    # Step3: "
    
    opener = urllib.request.build_opener()
    opener.addheaders = [('Authorization', 'Bearer {}'.format(token))]
    urllib.request.install_opener(opener)
    
    # Step4: Get JSON file"

    ladsweb_url = 'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/{}/{}/{:04d}/{:03d}.json'.format(collection, product_name,year,count_of_day)

    with urllib.request.urlopen(ladsweb_url) as url:
        data = json.loads(url.read().decode())   
    
    #pp = pprint.PrettyPrinter(indent=4)
    #pp.pprint(data)
    
    # Step5: Check if date is avaliable"
    
    file_name = ''
    for file in data['content']:
        #print(file['name'])
        file_name_prefix = '{}.A{:04d}{:03d}.{:02d}{:02d}'.format(product_name,year,count_of_day,hour,minute)
        if file_name_prefix in file['name']:
            file_name = file['name']      

    print(file_name)   

    # Step6: download the file"            
            
    if not path.exists(file_name):
        ladsweb_url = 'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/{}/{}/{:04d}/{:03d}/{}'.format(collection,product_name,year,count_of_day,file_name)
        urllib.request.urlretrieve(ladsweb_url, file_name) 
    
    return None

nasa_laads_daac_order_file("MYD021KM",61,2015,7,18,19,35)

MYD021KM.A2015199.1935.061.2018051071000.hdf


### Grabbing All Files for a Specific Day  <a class="anchor" id="download_files_script"></a>

Imagine you need all the data for a particular product on a specific day. Here's an example: downloading all NOAA VIIRS Active Fire data at 375-meter resolution for a single day:

In [19]:
%%time

from datetime import date
from os import path

import urllib.request
import urllib.request, json 
import pprint
import os

import socket

socket.setdefaulttimeout(10)

token = 'eyJ0eXAiOiJKV1QiLCJvcmlnaW4iOiJFYXJ0aGRhdGEgTG9naW4iLCJzaWciOiJlZGxqd3RwdWJrZXlfb3BzIiwiYWxnIjoiUlMyNTYifQ.eyJ0eXBlIjoiVXNlciIsInVpZCI6ImJtYXJjaGFudCIsImV4cCI6MTcyMjAzOTU1MiwiaWF0IjoxNzE2ODU1NTUyLCJpc3MiOiJFYXJ0aGRhdGEgTG9naW4ifQ.22EbP-kQdHunpQvnSB9eENEeV06J2lhCvliBIPo560VsUm4dBZeQEz-sE8n681zLKUERxOeXqyTWWYPGEtyJeMNuMfYkdAfFZN18VoU2SRG4W3YNanUXTjq15EONDAqyKr-UO-PcV5QKC_Xtkdj8nHlCwIkbzlZIBCBl1jZob0xmNM9qvCeqk0Rk8SXK80nvEz1Ji0fRwbuS_Cl4lgkNKZwMBNdziFP3rVClPews8waKoZmeu0zk20uylYazZY73QN8ZL6-vDYnSZL4Wo7RPsxornLg8wCD_xV4vv5PXzUe82VsthpyiYD03iu93MDOI9qX5lATChY8fo_4kjmkdNw'

output_dir = '/Volumes/HD15TB/Datasets/Research/NASA'

opener = urllib.request.build_opener()
opener.addheaders = [('Authorization', 'Bearer {}'.format(token))]
urllib.request.install_opener(opener)

year = 2023
month = 3
#day = 1

#platform = 'Suomi-NPP'
#instrument = 'VIIRS'
#product_category = 5111
#product = 'CLDPROP_L2_VIIRS_SNPP'

#platform = 'Suomi-NPP'
#instrument = 'VIIRS'
#product_category = 5110
#product = 'VNP02IMG'

#platform = 'Suomi-NPP'
#instrument = 'VIIRS'
#product_category = 5110
#product = 'VNP03IMG'

platform = 'Suomi-NPP'
instrument = 'VIIRS'
product_category = 5000
product = 'VNP14IMG'

#for day in [1,2,3,4,5,6,7,8,9]:
for day in [3]:
    
    error_log = []
    
    d0 = date(year, 1, 1)
    d1 = date(year, month, day)
    delta = d1 - d0

    count_of_day = delta.days + 1

    ladsweb_url = 'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/{}/{}/{:04d}/{:03d}.json'.format(product_category,product,year,count_of_day)

    print(ladsweb_url)

    with urllib.request.urlopen(ladsweb_url) as url:
        data = json.loads(url.read().decode())
    
    #pp = pprint.PrettyPrinter(indent=4)
    #pp.pprint(data['content'][:10])

    os.system( 'mkdir {}/{}'.format(output_dir,platform) )
    os.system( 'mkdir {}/{}/{}'.format(output_dir,platform,instrument) )
    os.system( 'mkdir {}/{}/{}/{}'.format(output_dir,platform,instrument,product))
    os.system( 'mkdir {}/{}/{}/{}/{:04d}'.format(output_dir,platform,instrument,product,year) )
    os.system( 'mkdir {}/{}/{}/{}/{:04d}/{:04d}_{:02d}_{:02d}'.format(output_dir,platform,instrument,product,year,year,month,day) )

    for file in data['content']:
        file_name = file['name']
        #print(file_name) 
        target_dir = '{}/{}/{}/{}/{:04d}/{:04d}_{:02d}_{:02d}/{}'.format(output_dir,platform,instrument,product,year,year,month,day,file_name)
        if not path.exists( target_dir ):
            try:
                ladsweb_url = 'https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/{}/{}/{:04d}/{:03d}/{}'.format(product_category,product,year,count_of_day,file_name)
                urllib.request.urlretrieve(ladsweb_url,target_dir) 
            except:
                print(file_name)
                print(ladsweb_url)
                print()
                error_log.append(file_name)

https://ladsweb.modaps.eosdis.nasa.gov/archive/allData/5000/VNP14IMG/2023/062.json


mkdir: /Volumes/HD15TB/Datasets/Research/NASA/Suomi-NPP: File exists
mkdir: /Volumes/HD15TB/Datasets/Research/NASA/Suomi-NPP/VIIRS: File exists
mkdir: /Volumes/HD15TB/Datasets/Research/NASA/Suomi-NPP/VIIRS/VNP14IMG: File exists
mkdir: /Volumes/HD15TB/Datasets/Research/NASA/Suomi-NPP/VIIRS/VNP14IMG/2023: File exists


CPU times: user 7.4 s, sys: 2.99 s, total: 10.4 s
Wall time: 3min 5s


## Links <a class="anchor" id="references"></a>

- [earthdata](https://www.earthdata.nasa.gov/learn/articles?page=7)
- [Migrating to the Cloud](https://disc.gsfc.nasa.gov/information/documents?title=Migrating%20to%20the%20Cloud#how-to)
- [cloud-tutorials](https://github.com/nasa/gesdisc-tutorials/tree/main/cloud-tutorials/notebooks)
- [NASA Active Fire Product](https://viirsland.gsfc.nasa.gov/Products/NASA/FireESDR.html)